lib/SPIRV/SPIRVReader.cpp

```diff
@@ -987,6 +995,48 @@ void SPIRVToLLVM::setLLVMLoopMetadata(const LoopInstType *LM,
   }
   if (LC & LoopControlNoFusionINTELMask)
     Metadata.push_back(getMetadataFromName("llvm.loop.fusion.disable"));
+  if (LC & spv::internal::LoopControlLoopCountINTELMask) {
+    // LoopCountINTELMask parameters are int64 and each parameter is stored
+    // as 2 SPIRVWords (int32)
+    uint64_t LoopCountMin =
+        static_cast<uint64_t>(LoopControlParameters[NumParam++]);
+    LoopCountMin |= static_cast<uint64_t>(LoopControlParameters[NumParam++])
+                    << 32;
+    if (static_cast<int64_t>(LoopCountMin) >= 0) {
+      Metadata.push_back(llvm::MDNode::get(
+          *Context,
+          getMetadataFromNameAndParameter("llvm.loop.intel.loopcount_min",
+                                          static_cast<int64_t>(LoopCountMin))));
+      assert(NumParam <= LoopControlParameters.size() &&
+             "Missing loop control parameter!");
+    }
+
+    uint64_t LoopCountMax =
+        static_cast<uint64_t>(LoopControlParameters[NumParam++]);
+    LoopCountMax |= static_cast<uint64_t>(LoopControlParameters[NumParam++])
+                    << 32;
+    if (static_cast<int64_t>(LoopCountMax) >= 0) {
+      Metadata.push_back(llvm::MDNode::get(
```

Comment:
You can update new overload of `getMetadataFromNameAndParameter` added by you to directly return `MDNode *` to simplify code here

Reply:
Updated in [1fbf78b](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1062/commits/1fbf78bf40440051bb7dbc9459d4cb1070ea8eae)